app/models/event.rb

```diff
@@ -106,4 +107,10 @@ def as_json(*)
   def schedule_notifications
     NotificationService.event_schedule_notifications(self)
   end
+
+  def destroy_event_users
```

Comment:
Doesn't the foreign key complain on this? Since it is done after the event itself is destroyed?